Storage/SQL/BrowserSchema.swift

```diff
@@ -1060,9 +1115,127 @@ open class BrowserTable: Table {
+
+        let tableListTableExists = sqliteMasterCursor[0] == 1
+        sqliteMasterCursor.close()
+
+        // If `tableList` still exists in this DB, then we need to continue to check if
+        // any table-specific migrations are required before removing it. Otherwise, if
+        // `tableList` does not exist, it is likely due to this being a brand new DB and
+        // no additional steps need to be taken at this point.
+        guard tableListTableExists else {
+            return true
+        }
+
+        // If we are unable to migrate the `clients` table from the schema table, we
+        // have failed and cannot continue.
+        guard migrateClientsTableFromSchemaTableIfNeeded(db) else {
+            return false
+        }
+
+        // Get the *previous* schema version (prior to v31) specified in `tableList`
+        // before dropping it.
+        let previousVersionCursor = db.executeQueryUnsafe("SELECT version FROM tableList WHERE name = 'BROWSER'", factory: IntFactory, withArgs: [] as Args)
+
+        let previousVersion = previousVersionCursor[0] ?? 0
+        previousVersionCursor.close()
+
+        // No other intermediary migrations are needed for the remaining tables and
+        // we have already captured the *previous* schema version specified in
+        // `tableList`, so we can now safely drop it.
+        log.info("Schema table migrations complete; Dropping 'tableList' table.")
+
+        let sql = "DROP TABLE IF EXISTS tableList"
+        if let err = db.executeChange(sql) {
+            log.error("Error dropping tableList table: \(err.localizedDescription)")
+            SentryIntegration.shared.sendWithStacktrace(message: "Error dropping tableList table: \(err.localizedDescription)", tag: "BrowserDB", severity: .error)
+            return false
+        }
+
+        // Lastly, write the *previous* schema version (prior to v31) to the database
+        // using `PRAGMA user_version = ?`.
+        if let err = db.setVersion(previousVersion) {
+            log.error("Error setting database version: \(err.localizedDescription)")
+            SentryIntegration.shared.sendWithStacktrace(message: "Error setting database version: \(err.localizedDescription)", tag: "BrowserDB", severity: .error)
+            return false
+        }
+        
         return true
     }
 
+    fileprivate func migrateClientsTableFromSchemaTableIfNeeded(_ db: SQLiteDBConnection) -> Bool {
+        // Query for the existence of the `clients` table to determine if we are
+        // migrating from an older DB version or if this is just a brand new DB.
+        let sqliteMasterCursor = db.executeQueryUnsafe("SELECT COUNT(*) AS number FROM sqlite_master WHERE type = 'table' AND name = '\(TableClients)'", factory: IntFactory, withArgs: [] as Args)
+        
+        let clientsTableExists = sqliteMasterCursor[0] == 1
+        sqliteMasterCursor.close()
+        
+        guard clientsTableExists else {
+            return true
```

Comment:
How about instead I break out the detection of whether or not we need to migrate in a separate function? That would clarify everything a bit better, I think.